gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -410,8 +410,13 @@ TypeCheckPattern::visit (HIR::QualifiedPathInExpression &pattern)
 void
 TypeCheckPattern::visit (HIR::ReferencePattern &pattern)
 {
-  rust_sorry_at (pattern.get_locus (),
-		 "type checking qualified path patterns not supported");
+  if (parent->get_kind () != TyTy::TypeKind::REF)
+    rust_error_at (pattern.get_locus (), "expected %s, found reference",
```

Comment:
You could add a dg-error test case for this

Reply:
Done 👍 please recheck